### ISSUE TITLE
[DynamoDBEnableServerSideEncryption] do not exist in the template

### ISSUE THREAD

**Issue by blazinaj** [aws-amplify/amplify-cli#2057]
**Describe the bug**
Getting `[DynamoDBEnableServerSideEncryption] do not exist in the template` error when I try to make any backend changes. I am using amplify cli version 1.11.0 and did the suggested fix from [Amplify Console aws-amplify/amplify-cli#35](https://github.com/aws-amplify/amplify-console/issues/35). Still cannot push any backend resources..

**To Reproduce**
1. Try to push any changes to existing Amplify Project after an Amplify release introduced a breaking change to the CloudFormation template.

**Expected behavior**
Be able to push backend resources.

**Screenshots**
![image](https://user-images.githubusercontent.com/39976117/63050292-98858a00-be8f-11e9-8f44-bf6ecc52afa0.png)
![image](https://user-images.githubusercontent.com/39976117/63050322-aaffc380-be8f-11e9-8f83-274fd6745ad3.png)

**Additional context**
This caused ALL of my Amplify Console branches to fail which had to be fixed one by one. Now I am trying to make a direct change to an Amplify CLI env and cannot get it to work at all.

**Comment by blazinaj** [comment]
In the console, we cannot get around this error. We applied the pre-build command like this and we are still getting the error.

![image](https://user-images.githubusercontent.com/39976117/63117677-22495c00-bf51-11e9-893f-322978d26204.png)


![image](https://user-images.githubusercontent.com/39976117/63117754-49079280-bf51-11e9-83d1-59669ceb4278.png)

**Comment by attilah** [comment]
@blazinaj I'd suggest that use the same version of the CLI what you've in the console. To try to solve the problem, remove the customization from the console and install 1.7.0 of CLI locally.

I think that your ```parameters``` file was touched by a newer version of the CLI or that parameter was added manually there.

The ```DynamoDBEnableServerSideEncryption``` parameter was added by [this](https://github.com/aws-amplify/amplify-cli/commit/baed927eec726190f3187d9967d6b4d5bc862fe0) commit 23 days ago, before this SSEEnabled flag was always true and the CLI had no way of configuring this behavior.

The problem you're facing is that the parameters section of the CloudFormation template is not in sync with the ```parameters.json``` file. The CLI supplies all the parameters from the json file, but CloudFormation cannot lookup the parameter itself in the template. If you remove it from the ```parameters.json``` the problem should be solved. Please verify it and get back to us.

**Comment by blazinaj** [comment]
Thank you for the response. Yes we are using a newer version of the CLI (1.11 || 1.12). 

This is causing all sorts of havoc in our development process (we run 5 envs concurrently with 3 developers), and we do not wish to roll back to the older version of the CLI. We are already utilizing the new mocking features.

Is there a bug report already made somewhere that I can follow for this?

**Comment by kaustavghosh06** [comment]
@blazinaj To understand this better, can you tell us whether in the cloudformation template located in `amplify/backend/api/<api-name>/build/loudformation-template.json` if you can the following block:

```json
"DynamoDBEnableServerSideEncryption": {
            "cType": "String",
            "Description": "Enable server side encryption powered by KMS.",
            "Default": "true",
            "AllowedValues": [
                "true",
                "false"
            ]
        },


```

**Comment by kaustavghosh06** [comment]
Closig this issue since the team wasn't able to reproduce this and lack of requester information.

**Comment by heitorlessa** [comment]
Bumped into this today in an existing stack, here are the steps to solve if anyone comes here later.

1. **Update API nested stack within the backend CloudFormation stack with `DynamoDBEnableServerSideEncryption: true`**

In my case, my API is named `apiawsserverlessairline` - Excerpt from: **amplify/backend/awscloudformation/nested-cloudformation-stack.yml**

```json
"apiawsserverlessairline": {
    "Type": "AWS::CloudFormation::Stack",
    "Properties": {
        "TemplateURL": "https://s3.amazonaws.com/awsserverlessairline-20190506121509-deployment/amplify-cfn-templates/api/cloudformation-template.json",
        "Parameters": {
            "AppSyncApiName": "awsserverlessairline",
            "DynamoDBBillingMode": "PAY_PER_REQUEST",
            "DynamoDBEnableServerSideEncryption": "true", <---
            ...
        }
    }
}
```

2. **Update API parameters.json file with `DynamoDBEnableServerSideEncryption: true`**

Excerpt from: **amplify/backend/api/awsserverlessairline/parameters.json**

{
    "AppSyncApiName": "awsserverlessairline",
    "DynamoDBBillingMode": "PAY_PER_REQUEST",
    "DynamoDBEnableServerSideEncryption": "true", <---
    "AuthCognitoUserPoolId": {
        "Fn::GetAtt": [
            "authcognitoa7d29d9d",
            "Outputs.UserPoolId"
        ]
    }
}

After that you can rerun `amplify push` or run through `Amplify Console` and that should work.

**Comment by roland-batovski** [comment]
I am running into the same issue right now. Updated my schema and the amplify push now fails every time with:

UPDATE_FAILED Answer AWS::CloudFormation::Stack Fri May 22 2020 09:28:17 GMT+0100 (British Summer Time) Parameters: [CreateAPIKey, DynamoDBEnableServerSideEncryption] do not exist in the template

This error message is thrown for every single one of my models. I tried updating Amplify to the latest version, with no success.
I checked the build folder for the required values in cloudformation-template.json and I was able to find them both:

"DynamoDBEnableServerSideEncryption": { "Type": "String", "Description": "Enable server side encryption powered by KMS.", "Default": "true", "AllowedValues": [ "true", "false" ] }

"CreateAPIKey": { "Type": "Number", "Description": "The boolean value to control if an API Key will be created or not. The value of the property is automatically set by the CLI. If the value is set to 0 no API Key will be created.", "Default": 0, "MinValue": 0, "MaxValue": 1 }

One thing that I noticed is that I couldn't find any of the above values in the amplify/backend/awscloudformation/nested-cloudformation-stack.yml file

I've already tried the fixes suggested above, with no luck. Any insight into this would be appreciated as it's currently stopping us from pushing any updates to our back-end

Cheers!
